stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/util/ListenerUtils.java

```diff
@@ -0,0 +1,267 @@
+import org.ballerinalang.jvm.scheduling.Scheduler;
+import org.ballerinalang.jvm.scheduling.Strand;
+import org.ballerinalang.jvm.values.MapValue;
+import org.ballerinalang.jvm.values.ObjectValue;
+import org.ballerinalang.messaging.rabbitmq.MessageDispatcher;
+import org.ballerinalang.messaging.rabbitmq.RabbitMQConnectorException;
+import org.ballerinalang.messaging.rabbitmq.RabbitMQConstants;
+import org.ballerinalang.messaging.rabbitmq.RabbitMQUtils;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Util class for RabbitMQ Listener actions handling.
+ *
+ * @since 1.0.4
+ */
+public class ListenerUtils {
+    private static final PrintStream console;
+    private static boolean started = false;
+    private static ArrayList<ObjectValue> services = new ArrayList<>();
+    private static BRuntime runtime;
+
+    public static void externInit(ObjectValue listenerObjectValue, Channel channel) {
+        listenerObjectValue.addNativeData(RabbitMQConstants.CHANNEL_NATIVE_OBJECT, channel);
+    }
+
+    public static Object registerListener(ObjectValue listenerObjectValue, ObjectValue service) {
+        runtime = BRuntime.getCurrentRuntime();
+        Channel channel = (Channel) listenerObjectValue.getNativeData(RabbitMQConstants.CHANNEL_NATIVE_OBJECT);
+        if (service != null) {
+            try {
+                declareQueueIfNotExists(service, channel);
+            } catch (IOException e) {
+                return RabbitMQUtils.returnErrorValue("I/O Error occurred while declaring the queue: " +
+                        e.getMessage());
+            }
+            if (isStarted()) {
+                services =
+                        (ArrayList<ObjectValue>) listenerObjectValue.getNativeData(RabbitMQConstants.CONSUMER_SERVICES);
+                startReceivingMessages(service, channel, listenerObjectValue);
+            }
+            services.add(service);
+            listenerObjectValue.addNativeData(RabbitMQConstants.CONSUMER_SERVICES, services);
+        }
+        return null;
+    }
+
+    public static Object start(ObjectValue listenerObjectValue) {
+        Strand strand = Scheduler.getStrand();
+        boolean autoAck;
+        ObjectValue channelObject = (ObjectValue) listenerObjectValue.get(RabbitMQConstants.CHANNEL_REFERENCE);
+        Channel channel = (Channel) listenerObjectValue.getNativeData(RabbitMQConstants.CHANNEL_NATIVE_OBJECT);
+        @SuppressWarnings(RabbitMQConstants.UNCHECKED)
+        ArrayList<ObjectValue> services =
+                (ArrayList<ObjectValue>) listenerObjectValue.getNativeData(RabbitMQConstants.CONSUMER_SERVICES);
+        @SuppressWarnings(RabbitMQConstants.UNCHECKED)
+        ArrayList<ObjectValue> startedServices =
+                (ArrayList<ObjectValue>) listenerObjectValue.getNativeData(RabbitMQConstants.STARTED_SERVICES);
+        if (services != null && !services.isEmpty()) {
```

Comment:
Better to invert the condition and exit early.

Reply:
Fixed